specification/postgresql/Microsoft.OrionDB.Management/models.tsp

```diff
@@ -0,0 +1,547 @@
+  /** Password authentication is disabled */
+  Disabled: "Disabled",
+}
+
+/**
+ * Data encryption type to depict if it is System Managed vs Azure Key vault.
+ */
+union DataEncryptionKeyType {
+  string,
+  /** System Managed data encryption */
+  SystemManaged: "SystemManaged",
+  /** Azure Key Vault data encryption */
+  AzureKeyVault: "AzureKeyVault",
+}
+
+/**
+ * Primary encryption key status for Data encryption enabled cluster.
+ */
+union KeyStatusEnum {
+  string,
+  /** Key is valid */
+  Valid: "Valid",
+  /** Key is invalid */
+  Invalid: "Invalid",
+}
+
+/**
+ * A value indicating whether Geo-Redundant backup is enabled on the cluster.
+ */
+union GeoRedundantBackupEnum {
+  string,
+  /** Geo-Redundant backup is enabled */
+  Enabled: "Enabled",
+  /** Geo-Redundant backup is disabled */
+  Disabled: "Disabled",
+}
+
+/**
+ * Public network access is enabled or not for Orion Db cluster.
+ */
+union ClusterPublicNetworkAccessState {
+  string,
+  /** Public network access is enabled */
+  Enabled: "Enabled",
+  /** Public network access is disabled */
+  Disabled: "Disabled",
+}
+
+/**
+ * The HA mode for the cluster.
+ */
+union HighAvailabilityMode {
+  string,
+  /** High availability is not enabled */
+  Disabled: "Disabled",
+  /** Zone redundant high availability is enabled*/
+  ZoneRedundant: "ZoneRedundant",
+  /** Same zone high availability is enabled*/
+  SameZone: "SameZone",
+}
+
+/**
+ * A state of a HA cluster that is visible to user.
+ */
+union ClusterHAState {
+  string,
+  /** High availability is not enabled */
+  NotEnabled: "NotEnabled",
+  /** Creating standby cluster */
+  CreatingStandby: "CreatingStandby",
+  /** Replicating data to standby */
+  ReplicatingData: "ReplicatingData",
+  /** Failing over to standby */
+  FailingOver: "FailingOver",
+  /** High availability is healthy */
+  Healthy: "Healthy",
+  /** Removing standby cluster */
+  RemovingStandby: "RemovingStandby",
+}
+
+/**
+ * Represents a Orion Db resource.
+ */
+model OrionDbResource is Azure.ResourceManager.TrackedResource<OrionDbResourceProperties> {
+  ...ResourceNameParameter<OrionDbResource, "orionDbClusterName", "clusters", "^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*">;
+
+  /**
+   * The SKU (pricing tier) of the Orion DB default pool.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "For backward compatibility"
+  defaultPoolSku?: Azure.ResourceManager.CommonTypes.Sku;
+
+  /**
+   * Describes the identity of the application.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "For backward compatibility"
+  identity?: Azure.ResourceManager.CommonTypes.UserAssignedIdentity;
+}
+
+/**
+ * The properties of a OrionDbResource.
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "For backward compatibility"
+model OrionDbResourceProperties {
+  /**
+   * The mode to create a new OrionDB cluster.
```

Comment:
"The mode to create a new OrionDB cluster." => "The mode used to determine if and how an OrionDB cluster is created or updated."